open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventConsumer.java

```diff
@@ -514,15 +509,52 @@ private void stopRunning()
     }
 
 
-    private class HandleRebalance implements ConsumerRebalanceListener
-    {
+    private class HandleRebalance implements ConsumerRebalanceListener {
         AuditLog auditLog = null;
+
         public HandleRebalance(AuditLog auditLog) {
             this.auditLog = auditLog;
         }
 
-        public void onPartitionsAssigned(Collection<TopicPartition> partitions)
-        {
+        public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+
+            // Check if we need to rewind to handle initial startup case -- but only on first assignment
+            try {
+                if (initialPartitionAssignment) {
+                    log.info("Received initial rebalance event");
+
+                    long partitionCount = partitions.size();
+
+                    if (partitionCount != 1) {
+                        log.info("Received rebalance event with " + partitionCount + " partitions.  This is not supported.");
+                    } else {
+                        // there is only one partition, so we can just grab the first one - and we'll try this once only
+                        initialPartitionAssignment = false;
```

Comment:
what happens if the partitions are revoked - should we set to true so that the partitions can be re assigned? Or does this only ever happen once as we should always have an offset (worth checking this is true after a revoke I think)?

Reply:
I don't think so. There is a lot more we may learn about how to respond to topology changes. I wanted to keep this simple.

The fix is for a very specific case - where we do not have an existing offset and are starting to read from a later point than we'd hoped. So I opted to only even check that on the first partition allocation, since after that point a offset should be known.

It's possible there may be timing windows where this isn't done, but the risk of making the behaviour worse is there, which is why I revert to leaving it to default if anything goes wrong. May not be perfect, but I think pragmatic